content/purgatorio/now-go-build/transfer-learning.md

```diff
@@ -1,5 +1,111 @@
+- [Conclusions](#Conclusions)
+
+##  What is Transfer Learning
+
+[Transfer Learning](https://en.wikipedia.org/wiki/Transfer_learning) is a research problem in Machine Learning that focuses on storing knowledge gained while solving one problem and applying it to a different but related problem. 
+
+This is done by _pre-training_ a model on a dataset (usually bigger and with generic content, like [ImageNet](http://www.image-net.org/)), and then _fine-tuning_ the model on a smaller and similar dataset.
+
+For example, you can fine-tune a pre-trained model from ImageNet on your much smaller dataset of images, in order to benefit the pre-training and obtain a series of almost-magical properties and results.
+
+To get a comprehensive overview of Why, What and How is used Transfer Learning, plus possible applications, read this awesome article:
+
+- [Transfer Learning - Machine Learning's Next Frontier](https://ruder.io/transfer-learning/index.html)
+
+
+#### Transfer Learning Magical Properties
+
+The theory of Transfer Learning suggests that when using a pre-trained model on a similar dataset:
+
+**You need less**:
+- Data: you need much smaller datasets to obtain the same (or better) model performances
+- Computational Power / Computational Time: empirical results have shown that pre-trained models reach top performances after much fewer epochs of training, that starting the training from scratch
+
+**You improve:**
+- Out-of-sample generalization: when predicting on test data you often obtain better accuracy, thus reducing overfitting (or whatever metric you're using)
+- Robustness: you make the model more "robust" to real-world low-quality data  
+
+So, **the first thing to do** every time you frame a new problem, is to ask yourself:
+
+**Can I leverage Transfer Learning in solving this problem?**
+
+
+## How Does Transfer Learning Work
+
+The practice of Transfer Learning allows reusing most of the parameters (weights) of a neural network previously trained on a problem similar to the one we have to solve, dwelling on the training only of the last layers that are usually those dedicated to the classification and/or regression of the features obtained with the previous layers.
+
+This allows us to obtain two key results:
+
+- reuse of the behavior of a network already trained to effectively **extract features** from input data
+- limit processing to a significantly **smaller number of parameters** (corresponding to the last layers)
+
+For example, if we were to classify apple varieties from an image, we could start by using a neural network already trained to classify images of planes, cars, dogs, cats, eggs, etc... This is because of the greater variety of training dataset ensures a better ability to extract features of various kinds from images.
+
+Of the _pre-trained neural network_, we would keep only the initial layers, going to redefine only the last layers of classification.
```

Comment:
Explain why last layers should not be used or removed - there is a reason for it, so that a specific purpose of TL is fulfilled as in the new replacing layers will be specific to the new dataset being used to train the model.